overlord/devicestate/handlers_remodel.go

```diff
@@ -111,6 +111,10 @@ func (m *DeviceManager) doSetModel(t *state.Task, _ *tomb.Tomb) (err error) {
 		return injectedSetModelError
 	}
 
+	if err := enforceValidationSetsForRemodel(st, new.ValidationSets()); err != nil {
+		return err
+	}
+
```

Comment:
if we error after here, we should put back in place the validation-sets setup that existed before this step, so that the reverting bits of the rest of the remodel have a chance work. We might need a test here or in managers_test.go about this scenarios failing so late in a remodel, where there is a different set of validation-sets setup before and after (if it had succeeded) the remodel that fails very late

Reply:
I've added a test to hopefully address this.